src/components/react-input-search/index.js

```diff
@@ -1,13 +1,53 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import './styles.css';
+
 const ReactInputSearch = (props) => {
   const [userInput, setUserInput] = useState('');
   const [filteredOptions, setFilteredOptions] = useState([]);
-  const [showOption, setShowOptions] = useState(false);
+  const [showOptions, setShowOptions] = useState(false);
+  const [activeItem, setActiveItem] = useState(-1);
+
+  const searchResultsNode = useRef();
+
+  useEffect(() => {
+    const handleMouseClick = (event) => {
+      if (searchResultsNode.current && !searchResultsNode.current.contains(event.target)) {
+        setActiveItem(-1);
+        setShowOptions(false);
+      }
+    };
+    document.addEventListener("mousedown", handleMouseClick);
+    return () => {
+      document.removeEventListener("mousedown", handleMouseClick);
+    };
+  }, []);
+
+  const onKeyDownHandler = ((event) => {
+    switch (event.key) {
+      case 'ArrowDown':
+        if (activeItem < (filteredOptions.length - 1)) {
+          setActiveItem(activeItem + 1);
+        }
+        break;
+      case 'ArrowUp':
+        if (activeItem > 0) {
+          setActiveItem(activeItem - 1);
+        }
+        break;
+      case 'Enter':
+        setShowOptions(false);
+        setUserInput(filteredOptions[activeItem].value);
+        setActiveItem(-1);
+        break;
+      default:
+        break;
+    }
+  });
```

Comment:
See if you can abstract this out into a custom hook as you can see the logic is becoming complex within this component